framework/include/mesh/MooseMesh.h

```diff
@@ -1171,6 +1195,9 @@ class MooseMesh : public MooseObject, public Restartable, public PerfGraphInterf
   /// Holds a map from subomdain ids to the boundary ids that are attached to it
   std::map<SubdomainID, std::set<BoundaryID>> _subdomain_boundary_ids;
```

Comment:
Last thing can you make these (for `_subdomain_boundary_ids` and `_neighbor_subdomain_boundary_ids` of type `std::unordered_map<SubdomainID, std::set<BoundaryID>>`.

Should only require a change here and a change in `getSubdomainInterfaceBoundaryIds()`.